TESTING.md

```diff
@@ -0,0 +1,137 @@
+# Tardigrade-ci Testing Documentation
+
+Testing tardigrade-ci implementation with various setup on local environment.
+
+## Testing
+
+This project can be utilized one of two ways, via docker or via a Makefile include.
+
+### Test Setup
+
+Configure or run the following script to setup a test project environment:
+
+```bash
+#!/bin/bash
+
+TESTDIR=$(mktemp --tmpdir -d tardigrade-ci.XXXXXXXXXX)
+cd $TESTDIR
+git init
+echo "FROM plus3it/tardigrade-ci:latest" > Dockerfile
+echo "SHELL := /bin/bash\n" > Makefile
+echo 'include $(shell test -f .tardigrade-ci || curl -sSL -o .tardigrade-ci "https://raw.githubusercontent.com/plus3it/tardigrade-ci/master/bootstrap/Makefile.bootstrap"; echo .tardigrade-ci)' >> Makefile
+echo '# tardigrade-ci' > .gitignore
+echo '.tardigrade-ci' >> .gitignore
+echo 'tardigrade-ci/' >> .gitignore
+
+docker pull python:3
+docker pull plus3it/tardigrade-ci:latest
+```
+
+### Via Docker (non-tardigrade-ci image)
+
+In a non-tardigrade-ci container, the first condition in `Makefile.bootstrap` ought to fail, resulting in the auto-init logic creating the `tardigrade-ci/` directory.
+
+1. Environment setup should contain `test-project/` described in the [Test Setup](#test-setup) section.
```

Comment:
Using the `TESTDIR` variable, this is no longer `test-project/`... It should be changed throughout the doc:

```suggestion
1. Environment setup should contain `"${TESTDIR:?}"` described in the [Test Setup](#test-setup) section.
```

Syntax: `"${...:?}"` checks that the var is set, non-empty and fails with a sensible message if not...

```
$ echo ${TESTDIR:?}
-bash: TESTDIR: parameter null or not set
```